### PR TITLE
GUACAMOLE-1721: Store only auth token from auth response in localStorage.

### PR DESCRIPTION
**Pull request by jmuehlner**
As discussed in [GUACAMOLE-1721](https://issues.apache.org/jira/browse/GUACAMOLE-1721), this change removes all auth response information except the auth token itself from local storage.

### CODE REVIEW

File: guacamole/src/main/frontend/src/app/auth/service/authenticationService.js [hunk 1]
```diff
@@ -84,12 +84,7 @@ angular.module('auth').factory('authenticationService', ['$injector',
             return cachedResult;
 
         // Return explicit null if no auth data is currently stored
-        var data = localStorageService.getItem(AUTH_STORAGE_KEY);
-        if (!data)
-            return null;
-
-        // Update cache and return retrieved auth result
-        return (cachedResult = new AuthenticationResult(data));
+        return null;
```
Comment (mike-jumper) [inline guacamole/src/main/frontend/src/app/auth/service/authenticationService.js:87]: As this function will now only return the in-memory cached auth result provided via `setAuthenticationResult()`, we should note this in the function's docs, as well as the fact that `setAuthenticationResult()` will be called automatically upon page load.

File: guacamole/src/main/frontend/src/app/auth/service/authenticationService.js [hunk 2]
```diff
@@ -115,9 +111,10 @@ angular.module('auth').factory('authenticationService', ['$injector',
             // Always store in cache
             cachedResult = data;
 
-            // Persist result past tab/window closure ONLY if not anonymous
+            // Persist auth token past tab/window closure ONLY if not anonymous
             if (data.username !== AuthenticationResult.ANONYMOUS_USERNAME)
-                localStorageService.setItem(AUTH_STORAGE_KEY, data);
+                localStorageService.setItem(
+                        AUTH_TOKEN_STORAGE_KEY, data.authToken);
```
Comment (mike-jumper) [inline guacamole/src/main/frontend/src/app/auth/service/authenticationService.js:117]: I think we should document that:

* Only the auth token is persisted locally, while the rest of the auth result is cached only in memory.
* The rest of the authentication result can be obtained by reauthenticating with the persisted token, which can be retrieved with `getCurrentToken()`.

**Review by mike-jumper — CHANGES_REQUESTED** [review]

File: guacamole/src/main/frontend/src/app/storage/services/localStorageService.js [hunk 3]
```diff
@@ -119,25 +140,56 @@ angular.module('storage').provider('localStorageService', [function localStorage
...
+     *
+     * @param {String} key
+     *     The arbitrary, unique key of the item to retrieve from localStorage.
+     *
+     * @returns {Object}
+     *     The value stored within localStorage under the given key,
+     *     automatically deserialized from JSON, or null if no such item is
+     *     present.
+     */
+    provider.getJsonItem = function getJsonItem(key) {
```
[the remark this answers is not in the document]
Comment (jmuehlner) [reply guacamole/src/main/frontend/src/app/storage/services/localStorageService.js:168]: Oh yeah, having the existing methods just handle both is a much better idea. I'll do that.

File: guacamole/src/main/frontend/src/app/auth/service/authenticationService.js (hunk 1 above)
Comment (jmuehlner) [reply guacamole/src/main/frontend/src/app/auth/service/authenticationService.js:87]: Will do

File: guacamole/src/main/frontend/src/app/auth/service/authenticationService.js (hunk 2 above)
Comment (jmuehlner) [reply guacamole/src/main/frontend/src/app/auth/service/authenticationService.js:125]: Done

File: guacamole/src/main/frontend/src/app/storage/services/localStorageService.js [hunk 4]
```diff
@@ -119,10 +122,18 @@ angular.module('storage').provider('localStorageService', [function localStorage
...
+        // Pull from internal storage, if present
         var data = storedItems[key];
-        if (data)
-            return JSON.parse(data);
+        if (data) {
+
+            // Serialize to JSON if possible
+            try {
+                return JSON.stringify(value);
+            } catch (ignore) {}
```
Comment (mike-jumper) [inline guacamole/src/main/frontend/src/app/storage/services/localStorageService.js:132]: I think you mean `JSON.parse()` here, but I also don't think we should attempt to parse values that we expect may not be JSON, as doing so introduces ambiguity (a value intended to be stored as a string may happen to also be valid JSON).

We could universally store everything as JSON, in which case a string would just be stored as a JSON string, or we could provide functions that are specifically intended for storage/retrieval of only string values.

**Review by mike-jumper — CHANGES_REQUESTED** [review]

File: guacamole/src/main/frontend/src/app/storage/services/localStorageService.js (hunk 4 above)
Comment (mike-jumper) [reply guacamole/src/main/frontend/src/app/storage/services/localStorageService.js:132]: > A raw string isn't a valid JSON-parseable value, ...

Isn't it? I get:

Stringifying and parsing the JavaScript string 'foo'

Comment (jmuehlner) [reply guacamole/src/main/frontend/src/app/storage/services/localStorageService.js:132]: Err, yeah. I'm not sure what was going on when I tested that earlier.

Comment (jmuehlner) [reply guacamole/src/main/frontend/src/app/storage/services/localStorageService.js]: Well, that was a bunch of ado about nothing. Removed my changes to this service.

**Pull request merged**